ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -60,6 +62,10 @@
  * will be accepted and all scheduled writes will be notified of the failure.
  */
 class NettyResponseChannel implements RestResponseChannel {
+  // add to this list if the connection needs to be closed on certain errors on GET, DELETE and HEAD.
+  // for a POST, we always close the connection on error because we expect the channel to be in a bad state.
+  static final List<HttpResponseStatus> CLOSE_CONNECTION_ERROR_STATUSES = new ArrayList<>();
```

Comment:
I think its useful because it is not trivial to find the piece of code that needs to be changed. This way, the logic is constant and clear and, I feel, easier to understand. Because what the code now says is "keep alive if not post and if im not supposed to close connection on this status"

I would prefer to retain this.